// validate-commit/dist/presets/angular.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var helpers_1 = require("../helpers");
exports.angular = {
    validate: function (message) {
        var MAX_LENGTH = 100;
        var PATTERN = /^(?:fixup!\s*)?(\w*)(\(([\w\$\.\*/-]*)\))?\: (.*)$/;
        var TYPES = {
            feat: true,
            fix: true,
            docs: true,
            style: true,
            refactor: true,
            perf: true,
            test: true,
            build: true,
            ci: true,
            chore: true,
            revert: true
        };
        message = message.split('\n').shift();
        if (message.length >= MAX_LENGTH) {
            helpers_1.log("Message is longer than " + MAX_LENGTH + " characters!", 'error');
            return false;
        }
        var match = PATTERN.exec(message);
        if (!match) {
            helpers_1.log("Message does not match \"<type>(<scope>): <subject>\"! was: " + message, 'error');
            return false;
        }
        var type = match[1];
        if (!TYPES.hasOwnProperty(type)) {
            helpers_1.log("'" + type + "' is not an allowed type!", 'error');
            helpers_1.log("Valid types are: " + Object.keys(TYPES).join(', '), 'info');
            return false;
        }
        return true;
    },
    ignorePattern: /^WIP\:/
};
